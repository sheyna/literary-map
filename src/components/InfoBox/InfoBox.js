import styles from './InfoBox.module.css';
import PropTypes from 'prop-types';

function InfoBox(props) {
	const {infoBoxLocation} = props;
	return (
		<section className={styles.InfoBox}>
			<h3>{infoBoxLocation.title}</h3>
			<p>{infoBoxLocation.townName}</p>
			<div 
				className={styles.close}
				onClick={() => {
					props.openInfoBox(false, {});
				}}
			>
				<i class="fas fa-times"></i>
			</div>
			<div className={styles.infoBoxLayout}>
				{ infoBoxLocation.image &&
					<div className={styles.infoBoxImg}>
						<img src={"img/" + infoBoxLocation.image} alt={infoBoxLocation.title}/>
					</div>
				}
				<div className={styles.infoBoxText}>
					<p>{infoBoxLocation.bodyText}</p>
					<p><span>Author(s):</span> {infoBoxLocation.author}</p>
					<p><span>Literary work(s):</span> {infoBoxLocation.book}</p>
				</div>
			</div>
		</section>
	)
}

InfoBox.propTypes = {
	infoBoxLocation: PropTypes.shape({
		title: PropTypes.string,
		book: PropTypes.string,
		author: PropTypes.string,
		thumb: PropTypes.string,
		image: PropTypes.string,
		positionTop: PropTypes.number,
		positionLeft: PropTypes.number,
		townName: PropTypes.string,
		bodyText: PropTypes.string,
	}),
	openInfoBox: PropTypes.func,
};

export default InfoBox;