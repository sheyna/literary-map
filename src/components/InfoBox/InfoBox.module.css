@keyframes loadBox {
	0% {height: 4rem;}
	66% {height: 40%;}
	100% {height: auto;}
}

@keyframes loadBoxMobile {
	0% {height: 4rem;}
	66% {height: 12%;}
	100% {height: auto;}
}

@keyframes slowDisplay {
	from {opacity: 0;}
	to {opacity: 1;}
}

.InfoBox {
	background-color: #FEFAF5;
	padding: .8rem;
	position: absolute;
	top: 9rem;
	left: 7%;
	z-index: 999;
	width: 85%;
}

.InfoBox h3 {
	font-size: 1rem;
	font-weight: 600;
	margin: 0 1rem .2rem 0;
}

.InfoBox p {
	margin: 0 0 .6rem;
}
.InfoBox p span {
	font-weight: 600;
	padding-right: .1rem;
}

.infoBoxLayout {
	display: flex;
	width: 100%;
	margin: 1rem 0 0;
	flex-direction: column;
	animation-name: slowDisplay;
  	animation-duration: 4s;
}

.infoBoxLayout .infoBoxImg + .infoBoxText {
	padding-left: 1rem;
}
.infoBoxLayout .infoBoxImg {
	flex: 0 0 auto;
}

.infoBoxLayout img {
	width: 100%;
}

.close {
	position: absolute;
	top: 0;
	right: 0;
	padding: .8rem;
	cursor: pointer;
}

@media screen and (min-width: 380px) {
	.InfoBox {
		top: 12rem;
		animation-name: loadBox;
  		animation-duration: 0.5s;
	}
}

@media screen and (min-width: 760px) {
	.infoBoxLayout {
		flex-direction: row;
	}
	.infoBoxLayout .infoBoxImg {
		flex: 0 0 15rem;
	}
	.InfoBox {
		margin: 0 0 3rem 0;
		position: fixed;
		bottom: 2rem;
		left: 10%;
		width: 80%;
		top: auto;
		animation-name: loadBoxMobile;
  		animation-duration: 0.5s;
	}
	.InfoBox h3 {
		font-size: 1.2rem;
	}
}

@media screen and (min-width: 1025px) {
	.InfoBox {
		margin: 2rem 0 2rem 0;
		width: 90%;
		position: relative;
		bottom: auto;
		left: auto;
		animation-name: loadBox;
  		animation-duration: 0.5s;
	}
	.infoBoxLayout .infoBoxText {
		overflow-y: scroll;
		max-height: 300px;
	}
}

@media screen and (min-width: 1025px) and (max-width: 1449px) {
	.infoBoxLayout .infoBoxImg {
		flex: 0 0 8rem;
	}
}

@media screen and (min-width: 1450px) {
	.InfoBox {
		margin: 0 0 3rem 0;
		width: 100%;
	}
	.infoBoxLayout .infoBoxText {
		max-height: 175px;
	}
}
