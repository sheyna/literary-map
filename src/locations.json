[
	{
		"locationId": 1,
		"title": "Brontë Country",
		"book": "Jane Eyre, Wuthering Heights ...",
		"author": "Emily, Charlotte and Anne Brontë",
		"thumb": "brontes-thumb.png",
		"image": "brontes.png",
		"positionTop": 30,
		"positionLeft": 49,
		"townName": "Haworth, West Yorkshire",
		"bodyText": "Among the most famous literary families the Brontë sisters wrote some of the most beloved works in literature. The parsonage where they lived with their father and brother is now a museum. The family, with the exception of Anne is buried in the church's vault. The Brontë Waterfall nearby was a favorite site of the sisters."
	},
	{
		"locationId": 2,
		"title": "A Romance of Exmoor",
		"book": "Lorna Doone",
		"author": "Richard Doddridge Blackmore",
		"thumb": "lorna-doone-thumb.png",
		"image": "lorna-doone.png",
		"positionTop": 77,
		"positionLeft": 27,
		"townName":  "Porlock, Somerset",
		"bodyText": "Porlock and Dulverton are the center of Lorna Doone Country. The Ship Inn in Portlock was well known to Coleridge, Wordsworth and Southery."
	},
	{
		"locationId": 3,
		"title": "The Bard of Stratford-upon-Avon",
		"book": "Hamlet, Romeo and Juliet, The Taming of the Shrew ...",
		"author": "William Shakespeare",
		"thumb": "shakespeare-thumb.png",
		"image": "shakespeare.png",
		"positionTop": 58,
		"positionLeft": 50,
		"townName":  "Stratford-upon-Avon, Warwickshire",
		"bodyText": "Widely believed to be the greatest writer in the English language, William Shakespeare wrote some 39 plays, 154 sonnets and three long narrative poems. The building believed to be his birthplace is a museum. His wife Anne Hathaway's home is nearby. "
	},
	{
		"locationId": 4,
		"title": "William Wordsworth's home",
		"book": "The Prelude, Ode: Intimations of Immortality, Ode to Duty, My Heart Leaps Up",
		"author": "William Wordsworth",
		"thumb": "wordsworth-thumb.png",
		"image": "wordsworth.png",
		"positionTop": 20,
		"positionLeft": 38,
		"townName":  "Grassmere, Cumbria",
		"bodyText": "Wordsworth wrote much of the poetry for which he is known as Dove Cottage near Grasmere. He lived here for 8 years with his sister, from 1799 to 1808."
	},
	{
		"locationId": 5,
		"title": "Count Dracula Arrives in England",
		"book": "Dracula",
		"author": "Bram Stoker",
		"thumb": "dracula-thumb.png",
		"image": "dracula.png",
		"positionTop": 22,
		"positionLeft": 60,
		"townName":  "Whitby, Yorkshire",
		"bodyText": "The ruins of Whitby Abbey stand on the cliff of above the town of Whitby overlooking the North Sea. Inspired by its gothic atmosphere when he visited in 1890, Bram Stocker set parts of his novel around the area. His Count Dracula comes to Whitby when he first arrives in England."
	},
	{
		"locationId": 6,
		"title": "London's Writers' Block",
		"book": "numerous",
		"author": "Henry James, Erskine Childers, T. S. Eliot, Somerset Maugham, Ian Fleming, George Elliot, Bram Stoker, Elizabeth Gaskell...",
		"thumb": "cheyne-thumb.png",
		"image": "cheyne.png",
		"positionTop":72,
		"positionLeft": 70,
		"townName":  "Cheyne Row, Chelsea, London",
		"bodyText": "This historic London 2-block street has been home to numerous authors. The Carlyle Mansions (number 24) has been nicknamed the 'Writers’ Block,' as it has been home to Henry James, Erskine Childers, T. S. Eliot, Somerset Maugham, Ian Fleming and other noted authors. Other authors have lived near by. George Elliot spent the last three weeks of her life at number 4. Bram Stoker lived at number 27. And Elizabeth Gaskell was born at number 93."
	},
	{
		"locationId": 7,
		"title": "Romance in the English Civil War",
		"book": "Peveril of the Peak",
		"author": "Sir Walter Scott",
		"thumb": "peveril-peak-thumb.png",
		"image": "peveril-peak.png",
		"positionTop": 37,
		"positionLeft": 52,
		"townName":  "Castleton, Derbyshire",
		"bodyText": "Now a ruins, this 11th-century castle, became the setting of one of Sir Walter Scott's Waverley novels (a series that includes Ivanhoe)."
	},
	{
		"locationId": 8,
		"title": "Visit's Gaskell's Cranford",
		"book": "Cranford",
		"author": "Elizabeth Gaskell",
		"positionTop": 40,
		"positionLeft": 46,
		"townName":  "Knutsford, Cheshire",
		"bodyText": "Elizabeth Gaskell's charming portrait of village life in the 19th century was based on her own childhood home of Knutsford."
	},
	{
		"locationId": 9,
		"title": "Elizabeth Gaskell's Home",
		"book": "Mary Barton, North and South, Cranford, Wives and Daughters",
		"author": "Elizabeth Gaskell",
		"positionTop": 38,
		"positionLeft": 48,
		"townName":  "Manchester",
		"bodyText": "Gaskell's historic home was restored in 2014. It is now a museum to the author and her times."
	},
	{
		"locationId": 10,
		"title": "Jane Austen's Home",
		"book": "Pride and Prejudice, Sense and Sensibility, Emma, Persuasion",
		"author": "Jane Austen",
		"positionTop": 78,
		"positionLeft": 62,
		"townName":  "Chawton, Hampshire",
		"bodyText": "Jane Austen spent the last eight years of her life at this home that is now a museum."
	},
	{
		"locationId": 11,
		"title": "Tolkien's Old Forest",
		"book": "Lord of the Rings",
		"author": "J.R.R. Tolkien",
		"positionTop": 55,
		"positionLeft": 49,
		"townName":  "Moseley, Worcestershire",
		"bodyText": "Moseley Bog is a woodland park in Moseley near Birmingham. J.R.R. Tolkien lived nearby as a boy. It is believed to have inspired the Old Forest in Middle Earth."
	},
	{
		"locationId": 12,
		"title": "Birthplace of Charles Dickens",
		"book": "Great Expectations, David Copperfield, Oliver Twist, A Tale of Two Cities ...",
		"author": "Charles Dickens",
		"positionTop": 83,
		"positionLeft": 62,
		"townName":  "Landport, Portsmouth, Hampshire",
		"bodyText": "The house where the famed novelist was born is now a museum. It is furnished in the Regency style popular at the time of his birth iin 1812."
	},
	{
		"locationId": 13,
		"title": "Hardy's Cottage",
		"book": "Far from the Madding Crowd, The Mayor of Casterbridge, Tess of the d'Urbervilles, and Jude the Obscure ",
		"author": "Thomas Hardy",
		"positionTop": 85,
		"positionLeft": 45,
		"townName":  "Dorchester, Dorset",
		"bodyText": "Thomas Hardy was born in a cottage outside Dorchester. He would later fictionalize Dorchester as the town of Casterbridge in his novel 'The Mayor of Casterbridge.' Hardy set most of his writing in a county he called Wessex. Hardy's Wessex encompassed the counties of Dorset, Wiltshire, Somerset, Devon, Hampshire and much of Berkshire."
	},
	{
		"locationId": 14,
		"title": "Jamaica Inn",
		"book": "Jamaica Inn",
		"author": "Daphne du Maurier",
		"positionTop": 87,
		"positionLeft": 17,
		"townName":  "Bolventor, Cornwall",
		"bodyText": "Daphne du Maurier made this inn from 1750 famous in her 1936 novel of the same name. The inn has a historical association with smuggling."
	},
	{
		"locationId": 15,
		"title": "Curious Affairs in Swindon",
		"book": "The Curious Incident of the Dog in the Night-Time and The Eyre Affair",
		"author": "Mark Haddon and Jasper Fforde",
		"positionTop": 70,
		"positionLeft": 53,
		"townName":  "Swindon, Wiltshire",
		"bodyText": "Christopher, the main character in Mark Haddon's 'The Curious Incident of the Dog in the Night-Time,' lives in Swindon. Swindon is also the location of Jasper Fforde's fantastical alternative 1985 of The Eyre Affair. The town planning office named streets in town after Fforde's characters. Look out for Thursday Street, Mycroft Road and Havisham Drive."
	},
	{
		"locationId": 16,
		"title": "Where did Louisa Musgrove fall?",
		"book": "Persuasion",
		"author": "Jane Austen",
		"positionTop": 85,
		"positionLeft": 39,
		"townName":  "The Cobb, Lyme Regis, Dorset",
		"bodyText": "Jane Austen sent her characters in her novel 'Persuasion' to Lyme to visit friends. It is there that the character Louisa Musgrove is injured after a fall at The Cobb."
	},
	{
		"locationId": 17,
		"title": "Deep in the Hundred Acre Wood",
		"book": " Winnie-the-Pooh and The House at Pooh Corner",
		"author": "A.A. Milne",
		"positionTop": 80,
		"positionLeft": 75,
		"townName":  "Five Hundred Acre Wood, Ashdown Forest, Hartfield, East Sussex",
		"bodyText": "From A.A. Milne's country home near Hartfield, the nearby Five Hundred Acre Wood would serve as the inspiration for the fictional wood inhabited by Winnie the Pooh and his friends. Many features from the stories can be found in this forest. An interpretive trail provides insights into this classic story."
	}
]
