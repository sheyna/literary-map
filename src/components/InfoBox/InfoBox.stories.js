import React from 'react';
import InfoBox from './InfoBox';
import locations from '../../locations.json';
const openInfoBox = () => {};

export default {
  title: 'component/InfoBox',
  component: InfoBox,
}

export const textNoImage = () => <InfoBox 
  infoBoxLocation={locations[7]}
  openInfoBox={openInfoBox}
/>

export const textWithImage = () => <InfoBox 
  infoBoxLocation={locations[0]}
  openInfoBox={openInfoBox}
/>