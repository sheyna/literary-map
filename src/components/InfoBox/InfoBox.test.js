import ReactTestRenderer from 'react-test-renderer';
import InfoBox from  './InfoBox';
import locations from '../../locations.json';

describe('InfoBox component', () => {
  it('should match snapshot', () => {
    const component = ReactTestRenderer.create(
      <InfoBox infoBoxLocation={locations[0]} openInfoBox={true}/>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});