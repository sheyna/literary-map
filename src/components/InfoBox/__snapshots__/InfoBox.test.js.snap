// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoBox component should match snapshot 1`] = `
<section
  className="InfoBox"
>
  <h3>
    Brontë Country
  </h3>
  <p>
    Haworth, West Yorkshire
  </p>
  <div
    className="close"
    onClick={[Function]}
  >
    <i
      class="fas fa-times"
    />
  </div>
  <div
    className="infoBoxLayout"
  >
    <div
      className="infoBoxImg"
    >
      <img
        alt="Brontë Country"
        src="img/brontes.png"
      />
    </div>
    <div
      className="infoBoxText"
    >
      <p>
        Among the most famous literary families the Brontë sisters wrote some of the most beloved works in literature. The parsonage where they lived with their father and brother is now a museum. The family, with the exception of Anne is buried in the church's vault. The Brontë Waterfall nearby was a favorite site of the sisters.
      </p>
      <p>
        <span>
          Author(s):
        </span>
         
        Emily, Charlotte and Anne Brontë
      </p>
      <p>
        <span>
          Literary work(s):
        </span>
         
        Jane Eyre, Wuthering Heights ...
      </p>
    </div>
  </div>
</section>
`;
